### PR TITLE
Feat/255 select person responsible

### PR DESCRIPTION
**Pull request by ayeshmcg**
Card: https://github.com/bcgov/cas-reporting/issues/255
Changes:

Added Person Resposible Page
Added Person Resposible Page
Added Model for report person responsible
Added API
Added check for opertation Type, if LFO wil take to a TBD page othewise  to facility review page

To test:-
Go to page /reporting/reports after logging in with BCeId
Select any operation
It wil open operation reveiw information page and then click on save and Coninue, it wil take you to /reporting/reports/{report_version}/person-responsible page
If there are any previously selected person resposible for the report it will pre-populate the contact details, if not select any contact and save.

### CODE REVIEW

**Review by dleard — CHANGES_REQUESTED** [review]
Tested it locally & it works great! A few minor changes needed & then this should be good to go! :100:

File: bciers/apps/reporting/src/app/bceidbusiness/industry_user_admin/reports/[version_id]/person-responsible/page.tsx [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+import Loading from "@bciers/components/loading/SkeletonForm";
 
-export default async function Page() {
-  return <PersonResponsible />;
+export default async function Page({
+  params,
+}: {
+  params: { version_id: number };
+}) {
+  console.log("versionId", params.version_id);
```
[the remark this answers is not in the document]
Comment (ayeshmcg) [reply bciers/apps/reporting/src/app/bceidbusiness/industry_user_admin/reports/[version_id]/person-responsible/page.tsx:10]: missed that,

File: bciers/apps/reporting/src/app/components/operations/personResponsible/PersonResponsible.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    null,
+  );
+  const [contactFormData, setContactFormData] = useState<any>(null);
+  const [formData, setFormData] = useState({
+    person_responsible: "", // Default to empty string
+  });
+
+  const [schema, setSchema] = useState<RJSFSchema>(personResponsibleSchema);
+  const router = useRouter();
+  const saveAndContinueUrl = `/reports/${version_id}/review-facilities`;
```
[the remark this answers is not in the document]
Comment (ayeshmcg) [reply bciers/apps/reporting/src/app/components/operations/personResponsible/PersonResponsible.tsx:59]: So we need to get the facilityId for this , right?

[the remark this answers is not in the document]
Comment (dleard) [reply bciers/apps/reporting/src/app/components/operations/personResponsible/PersonResponsible.tsx:59]: Yeah, it'll have to be fetched from the report_version

File: bc_obps/service/report_service.py [hunk 3]
```diff
@@ -103,3 +106,33 @@ def save_report_operation(cls, report_version_id: int, data: ReportOperationIn)
...
+                'position_title': data.position_title,
+                'business_role': data.business_role,
+                'street_address': data.street_address,
+                'municipality': data.municipality,
+                'province': data.province,
+                'postal_code': data.postal_code,
+            },
+        )
+
+        return cast(ReportPersonResponsibleOut, report_person_responsible)
```
[the remark this answers is not in the document]
Comment (pbastia) [reply bc_obps/service/report_service.py:138]: nope! Django-ninja does this for us, it's magic!

File: bc_obps/service/report_service.py [hunk 4]
```diff
@@ -103,3 +106,33 @@ def save_report_operation(cls, report_version_id: int, data: ReportOperationIn)
         report_operation.save()
 
         return report_operation
+
```
[the remark this answers is not in the document]
Comment (ayeshmcg) [reply bc_obps/service/report_service.py:109]: @dleard What about creating report_person_responsible_service?

**Comment by ayeshmcg** [comment]
@dleard  Added check for operation type as well, if LFO-> it will show a page with TBD

File: bc_obps/reporting/api/facility_report.py
Comment (dleard) [inline bc_obps/reporting/api/facility_report.py:1]: Looks like this file got messed up a little bit during a rebase. The activity-list endpoint was removed & there are some regressions with the return types/values for get/save facility_report

**Review by dleard — CHANGES_REQUESTED** [review]

File: bc_obps/reporting/api/report_person_responsible.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+@router.post(
+    "/report-version/{version_id}/report-contact",
+    response={201: ReportPersonResponsibleOut, custom_codes_4xx: Message},
+    tags=EMISSIONS_REPORT_TAGS,
+    description=(
+        """Creates or updates a contact associated with a report version.
+                    Includes fields like legal name, trade name, operation details, and contact information."""
+    ),
+    auth=authorize("approved_authorized_roles"),
```
Comment (dleard) [inline bc_obps/reporting/api/report_person_responsible.py:37]: This permissions should be scoped to the `approved_industry_user` as an idir user should not be able to mutate the contents of a report.
```suggestion
    auth=authorize("approved_industry_user"),
```

File: bc_obps/reporting/api/reports.py [hunk 6]
```diff
@@ -50,7 +51,7 @@ def get_report_operation_by_version_id(
     tags=EMISSIONS_REPORT_TAGS,
     description="""Updates given report operation with fields: Operator Legal Name, Operator Trade Name, Operation Name, Operation Type,
     Operation BC GHG ID, BC OBPS Regulated Operation ID, Operation Representative Name, and Activities.""",
-    auth=authorize("approved_industry_user"),
+    auth=authorize("approved_authorized_roles"),
```
Comment (dleard) [inline bc_obps/reporting/api/reports.py:54]: Same thing here
```suggestion
    auth=authorize("approved_industry_user"),
```

File: bc_obps/reporting/models/report_person_responsible.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+from django.db import models
+
+from registration.models.time_stamped_model import TimeStampedModel
+from registration.models.user_and_contact_common_info import UserAndContactCommonInfo
+from reporting.models.report_version import ReportVersion
+
+
+class ReportPersonResponsible(UserAndContactCommonInfo, TimeStampedModel):
+    # A report version can have multiple persons responsible
```
Comment (dleard) [inline bc_obps/reporting/models/report_person_responsible.py:9]: Is this comment correct? I thought a report version can only have one person responsible?

File: bciers/apps/reporting/src/app/components/operations/personResponsible/PersonResponsible.tsx [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        contactFormData,
+      );
+      setSchema(updatedSchema);
+    }
+  }, [selectedContactId, contactFormData]);
+
+  useEffect(() => {
+    const getFacilityId = async () => {
+      const facilityReport = await getFacilityReport(version_id);
+      console.log("facilityReport", facilityReport);
```
Comment (dleard) [inline bciers/apps/reporting/src/app/components/operations/personResponsible/PersonResponsible.tsx:130]: Remove debugging line

**Review by dleard — APPROVED** [review]
Really nice work! A couple minor things around permissions to fix up and then this is good to merge

**Pull request merged**